src/isar/state_machine/transitions/mission.py

```diff
@@ -67,30 +70,42 @@ def get_mission_transitions(state_machine: "StateMachine") -> List[dict]:
         {
             "trigger": "resume",
             "source": state_machine.paused_state,
+            "dest": state_machine.resuming_state,
+            "conditions": def_transition(state_machine, trigger_resume_mission_event),
+        },
+        {
+            "trigger": "mission_resumed",
+            "source": state_machine.resuming_state,
             "dest": state_machine.monitor_state,
-            "conditions": def_transition(state_machine, resume_mission),
         },
         {
-            "trigger": "resume",
-            "source": state_machine.paused_state,
-            "dest": state_machine.paused_state,
+            "trigger": "mission_resuming_failed",
+            "source": state_machine.resuming_state,
+            "dest": state_machine.await_next_mission_state,
+            "conditions": def_transition(state_machine, resume_mission_failed),
         },
         {
             "trigger": "resume",
             "source": state_machine.return_home_paused_state,
+            "dest": state_machine.resuming_return_home_state,
+            "conditions": def_transition(state_machine, trigger_resume_mission_event),
+        },
+        {
+            "trigger": "return_home_mission_resumed",
+            "source": state_machine.resuming_return_home_state,
             "dest": state_machine.returning_home_state,
-            "conditions": def_transition(state_machine, resume_mission),
         },
         {
-            "trigger": "resume",
-            "source": state_machine.return_home_paused_state,
-            "dest": state_machine.return_home_paused_state,
+            "trigger": "return_home_mission_resuming_failed",
+            "source": state_machine.resuming_return_home_state,
+            "dest": state_machine.await_next_mission_state,
+            "before": def_transition(state_machine, resume_mission_failed),
         },
         {
             "trigger": "resume_lockdown",
             "source": state_machine.return_home_paused_state,
             "dest": state_machine.going_to_lockdown_state,
-            "conditions": def_transition(state_machine, resume_mission),
+            "conditions": def_transition(state_machine, trigger_resume_mission_event),
```

Comment:
Can we handle if the resume fails? If it is complicated to accomplish here, we could potentially not consume the lockdown request until we've managed to resume the return home mission, then we can easily transition from returning_home to going_to_lockdown